### DISCUSSION TITLE
Mixing vmware and bare metal

### DISCUSSION THREAD

**Discussion by dkoci** [okd-project/okd#851] [Q&A]
Hi. I have my okd cluster running in vmware (UPI). Is it possible to add few worker nodes, which would be running on physical servers instead vmware?

**Comment by vrutkovs** [comment]
>Is it possible to add few worker nodes

Yes, but you might get errors when vSphere features are used - like volumes

**Reply by dkoci** [reply]
All volumes are mounted to external NFS. What about worker.ign file? Can I use existing one which I used to create workers in vmware?

**Reply by vrutkovs** [reply]
New worker.ign is not required afaik, it doesn't have any vSphere specifics. You might see non-critical errors in kubelet though

**Reply by rvanderp3** [reply]
It's not recommended to mix infrastructure types as many components expect all nodes to be enabled with the declared cloud provider.  My recommendation would be to perform a bare metal install on VMware if you don't need any of the VMware features.
